core/test/network-records-to-devtools-log.js

```diff
@@ -227,6 +244,48 @@ function addRedirectResponseIfNeeded(networkRecords, record) {
 function networkRecordsToDevtoolsLog(networkRecords, options = {}) {
   const devtoolsLog = [];
   networkRecords.forEach((networkRecord, index) => {
+    // If we're operating on network record raw objects (not NetworkRequest instances),
+    // then we're operating on test data that may need to be massaged a bit.
+    if (networkRecord.constructor === Object) {
+      // Temporary code while we transition away from startTime and endTime.
+      // This allows us to defer slightly changes to test files.
+      // TODO: remove after timing refactor is done
+      // See https://github.com/GoogleChrome/lighthouse/pull/14311
+      if (networkRecord.startTime !== undefined) {
+        networkRecord.networkRequestTime = networkRecord.startTime;
+        // Old tests never distinguished between these two.
+        networkRecord.rendererStartTime = networkRecord.startTime;
+      }
+      if (networkRecord.endTime !== undefined) {
+        networkRecord.networkEndTime = networkRecord.endTime;
+      }
+      if (networkRecord.responseReceivedTime !== undefined) {
+        networkRecord.responseHeadersEndTime = networkRecord.responseReceivedTime;
+      }
+
+      // Set timing.requestTime and timing.receiveHeadersEnd to be values that
+      // NetworkRequest will pull from for networkRequestTime and responseHeadersEndTime,
+      // so this roundtrips correctly. Unless, of course, timing values are explicitly set
+      // already.
+
+      // If networkRecord.timing explicitly does not having a timing object, only create one
+      // if we absolutely need it. See _recomputeTimesWithResourceTiming.
+      const netReqTime = networkRecord.networkRequestTime;
+      const willNeedTimingObject =
+        (netReqTime !== undefined && netReqTime !== networkRecord.rendererStartTime) ||
+        (networkRecord.responseHeadersEndTime !== undefined);
+      if (willNeedTimingObject) networkRecord.timing = networkRecord.timing || {};
+
+      if (networkRecord.timing) {
+        if (networkRecord.timing.requestTime === undefined) {
```

Comment:
don't these both need to be set if one of them is going to be?

Reply:
Yeah, but the idea here was to never override a provided timing object, in case a test defined one (even partially).

Maybe instead this should check if both these timing fields are undefined before assigning either?